### ISSUE TITLE
[MP Server] Parsing error loading recipe

### ISSUE THREAD

**Issue by kalinzange** [NoComment1105/DeepslateCutting#2]
### What happened?

When loading the server I get Parsing error loading recipe for deepslate blocks

### Screenshots

N/A

### Relevant log output

```shell
[03:34:42] [Worker-Main-7/ERROR]: Parsing error loading recipe deepslatecutting:deepslate_slab
com.google.gson.JsonSyntaxException: Unknown item 'deepslatecutting:deepslate_slab'
	at net.minecraft.class_1869.method_17878(class_1869.java:253) ~[intermediary-server.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_1869.method_8155(class_1869.java:253) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869.method_35228(class_1869.java:240) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869$class_1870.method_8164(class_1869.java:273) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869$class_1870.method_8121(class_1869.java:260) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_17720(class_1863.java:123) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_20705(class_1863.java:54) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_18788(class_1863.java:35) ~[intermediary-server.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) [?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) [?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1769) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1756) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?]
[03:34:42] [Worker-Main-7/ERROR]: Parsing error loading recipe deepslatecutting:deepslate_stairs
com.google.gson.JsonSyntaxException: Unknown item 'deepslatecutting:deepslate_stairs'
	at net.minecraft.class_1869.method_17878(class_1869.java:253) ~[intermediary-server.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_1869.method_8155(class_1869.java:253) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869.method_35228(class_1869.java:240) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869$class_1870.method_8164(class_1869.java:273) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869$class_1870.method_8121(class_1869.java:260) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_17720(class_1863.java:123) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_20705(class_1863.java:54) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_18788(class_1863.java:35) ~[intermediary-server.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) [?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) [?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1769) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1756) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?]
[03:34:42] [Worker-Main-7/ERROR]: Parsing error loading recipe deepslatecutting:deepslate_wall
com.google.gson.JsonSyntaxException: Unknown item 'deepslatecutting:deepslate_wall'
	at net.minecraft.class_1869.method_17878(class_1869.java:253) ~[intermediary-server.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_1869.method_8155(class_1869.java:253) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869.method_35228(class_1869.java:240) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869$class_1870.method_8164(class_1869.java:273) ~[intermediary-server.jar:?]
	at net.minecraft.class_1869$class_1870.method_8121(class_1869.java:260) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_17720(class_1863.java:123) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_20705(class_1863.java:54) ~[intermediary-server.jar:?]
	at net.minecraft.class_1863.method_18788(class_1863.java:35) ~[intermediary-server.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) [?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) [?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1769) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1756) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?]
[03:34:42] [Worker-Main-7/INFO]: Loaded 74 recipes
[03:34:42] [Worker-Main-7/ERROR]: Couldn't parse loot table deepslatecutting:blocks/deepslate_slab
com.google.gson.JsonSyntaxException: Expected name to be an item, was unknown string 'deepslatecutting:deepslate_slab'
	at net.minecraft.class_3518.method_17995(class_3518.java:127) ~[intermediary-server.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_3518.method_15256(class_3518.java:127) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15288(class_3518.java:135) ~[intermediary-server.jar:?]
	at net.minecraft.class_77$class_78.method_413(class_77.java:55) ~[intermediary-server.jar:?]
	at net.minecraft.class_77$class_78.method_443(class_77.java:40) ~[intermediary-server.jar:?]
	at net.minecraft.class_85$class_90.method_441(class_85.java:151) ~[intermediary-server.jar:?]
	at net.minecraft.class_85$class_90.method_424(class_85.java:129) ~[intermediary-server.jar:?]
	at net.minecraft.class_79$class_5337.method_29319(class_79.java:86) ~[intermediary-server.jar:?]
	at net.minecraft.class_79$class_5337.method_517(class_79.java:74) ~[intermediary-server.jar:?]
	at net.minecraft.class_5330$class_5333.deserialize(class_5330.java:102) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[server.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[server.jar:?]
	at net.minecraft.class_3518.method_15291(class_3518.java:440) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15272(class_3518.java:448) ~[intermediary-server.jar:?]
	at net.minecraft.class_55$class_57.method_358(class_55.java:173) ~[intermediary-server.jar:?]
	at net.minecraft.class_55$class_57.deserialize(class_55.java:169) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[server.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[server.jar:?]
	at net.minecraft.class_3518.method_15291(class_3518.java:440) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15283(class_3518.java:456) ~[intermediary-server.jar:?]
	at net.minecraft.class_52$class_54.method_340(class_52.java:213) ~[intermediary-server.jar:?]
	at net.minecraft.class_52$class_54.deserialize(class_52.java:209) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:925) ~[server.jar:?]
	at net.minecraft.class_60.method_20711(class_60.java:44) ~[intermediary-server.jar:?]
	at java.util.HashMap.forEach(HashMap.java:1425) ~[?:?]
	at net.minecraft.class_60.method_20712(class_60.java:42) ~[intermediary-server.jar:?]
	at net.minecraft.class_60.method_18788(class_60.java:17) ~[intermediary-server.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:610) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) [?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) [?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1769) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1756) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?]
[03:34:42] [Worker-Main-7/ERROR]: Couldn't parse loot table deepslatecutting:blocks/deepslate_wall
com.google.gson.JsonSyntaxException: Expected name to be an item, was unknown string 'deepslatecutting:deepslate_wall'
	at net.minecraft.class_3518.method_17995(class_3518.java:127) ~[intermediary-server.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_3518.method_15256(class_3518.java:127) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15288(class_3518.java:135) ~[intermediary-server.jar:?]
	at net.minecraft.class_77$class_78.method_413(class_77.java:55) ~[intermediary-server.jar:?]
	at net.minecraft.class_77$class_78.method_443(class_77.java:40) ~[intermediary-server.jar:?]
	at net.minecraft.class_85$class_90.method_441(class_85.java:151) ~[intermediary-server.jar:?]
	at net.minecraft.class_85$class_90.method_424(class_85.java:129) ~[intermediary-server.jar:?]
	at net.minecraft.class_79$class_5337.method_29319(class_79.java:86) ~[intermediary-server.jar:?]
	at net.minecraft.class_79$class_5337.method_517(class_79.java:74) ~[intermediary-server.jar:?]
	at net.minecraft.class_5330$class_5333.deserialize(class_5330.java:102) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[server.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[server.jar:?]
	at net.minecraft.class_3518.method_15291(class_3518.java:440) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15272(class_3518.java:448) ~[intermediary-server.jar:?]
	at net.minecraft.class_55$class_57.method_358(class_55.java:173) ~[intermediary-server.jar:?]
	at net.minecraft.class_55$class_57.deserialize(class_55.java:169) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[server.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[server.jar:?]
	at net.minecraft.class_3518.method_15291(class_3518.java:440) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15283(class_3518.java:456) ~[intermediary-server.jar:?]
	at net.minecraft.class_52$class_54.method_340(class_52.java:213) ~[intermediary-server.jar:?]
	at net.minecraft.class_52$class_54.deserialize(class_52.java:209) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:925) ~[server.jar:?]
	at net.minecraft.class_60.method_20711(class_60.java:44) ~[intermediary-server.jar:?]
	at java.util.HashMap.forEach(HashMap.java:1425) ~[?:?]
	at net.minecraft.class_60.method_20712(class_60.java:42) ~[intermediary-server.jar:?]
	at net.minecraft.class_60.method_18788(class_60.java:17) ~[intermediary-server.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:610) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) [?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) [?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1769) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1756) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?]
[03:34:42] [Worker-Main-7/ERROR]: Couldn't parse loot table deepslatecutting:blocks/deepslate_stairs
com.google.gson.JsonSyntaxException: Expected name to be an item, was unknown string 'deepslatecutting:deepslate_stairs'
	at net.minecraft.class_3518.method_17995(class_3518.java:127) ~[intermediary-server.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_3518.method_15256(class_3518.java:127) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15288(class_3518.java:135) ~[intermediary-server.jar:?]
	at net.minecraft.class_77$class_78.method_413(class_77.java:55) ~[intermediary-server.jar:?]
	at net.minecraft.class_77$class_78.method_443(class_77.java:40) ~[intermediary-server.jar:?]
	at net.minecraft.class_85$class_90.method_441(class_85.java:151) ~[intermediary-server.jar:?]
	at net.minecraft.class_85$class_90.method_424(class_85.java:129) ~[intermediary-server.jar:?]
	at net.minecraft.class_79$class_5337.method_29319(class_79.java:86) ~[intermediary-server.jar:?]
	at net.minecraft.class_79$class_5337.method_517(class_79.java:74) ~[intermediary-server.jar:?]
	at net.minecraft.class_5330$class_5333.deserialize(class_5330.java:102) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[server.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[server.jar:?]
	at net.minecraft.class_3518.method_15291(class_3518.java:440) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15272(class_3518.java:448) ~[intermediary-server.jar:?]
	at net.minecraft.class_55$class_57.method_358(class_55.java:173) ~[intermediary-server.jar:?]
	at net.minecraft.class_55$class_57.deserialize(class_55.java:169) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[server.jar:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[server.jar:?]
	at net.minecraft.class_3518.method_15291(class_3518.java:440) ~[intermediary-server.jar:?]
	at net.minecraft.class_3518.method_15283(class_3518.java:456) ~[intermediary-server.jar:?]
	at net.minecraft.class_52$class_54.method_340(class_52.java:213) ~[intermediary-server.jar:?]
	at net.minecraft.class_52$class_54.deserialize(class_52.java:209) ~[intermediary-server.jar:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:952) ~[server.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:925) ~[server.jar:?]
	at net.minecraft.class_60.method_20711(class_60.java:44) ~[intermediary-server.jar:?]
	at java.util.HashMap.forEach(HashMap.java:1425) ~[?:?]
	at net.minecraft.class_60.method_20712(class_60.java:42) ~[intermediary-server.jar:?]
	at net.minecraft.class_60.method_18788(class_60.java:17) ~[intermediary-server.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:610) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:71) ~[intermediary-server.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:70) ~[intermediary-server.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) [?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) [?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1769) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1756) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?]
```

### DeepslateCutting and Minecraft Version

DeepslateCutting-1.1.2 for Minecraft 1.17.1

### Operating System

Windows Server 2019 (Version 1809)

**Comment by NoComment1105** [comment]
Thanks for the report. This is know to me and I don't know how to fix it yet. I just chose ot push the release and the spam wasn't large enough to cause an issue in my test. I will look into fix it soon

**Comment by gniftygnome** [comment]
I think the log spam only occurs when the mod is not "enabled" in its configuration.  Probably the issue happens because the recipes are loaded without the blocks they reference being added.  You might find Draylar's [maybe-data](https://github.com/Draylar/maybe-data) useful for this (I think LibCD would also be needed for this use case ... or you could borrow Draylar's implementation of the conditional recipe loader).

**Comment by NoComment1105** [comment]
> I think the log spam only occurs when the mod is not "enabled" in its configuration. Probably the issue happens because the recipes are loaded without the block they reference being added.

This is indeed the case, thank you for your reply.
Also thank you for maybe-data, I will look into that and see if I can utilise it 👍🏼

**Comment by NoComment1105** [comment]
I just looked into maybe-data but unfortunately that library only loads recipes based on a mod being loaded, not a block being registered. Which is unfortunate. Ill try and make contact with the mod author and see what can be done about it

**Comment by gniftygnome** [comment]
If I'm reading the source correctly (and I certainly might be missing something ... haven't mocked up a mod to test this) maybe-data looks like if instead of a "modid" field in the recipe JSON you put an "item" field it will load based on whether the item exists.

It reads from the JSON like this:
```java
                Condition condition = GSON.fromJson(obj.get("condition"), Condition.class);
                if(condition.verify()) {
                    valid.put(new Identifier(identifier.getNamespace(), String.format("mayberecipe_%s", identifier.getPath())), obj.get("recipe"));
                }
```

And condition.verify() has this:

```java
    public boolean verify() {
        if(!modid.isEmpty()) {
            return FabricLoader.getInstance().isModLoaded(modid);
        }

        if(!item.isEmpty()) {
            return Registry.ITEM.getIds().contains(new Identifier(item));
        }

        return false;
    }
```

So I think it has undocumented functionality to check by item existence instead of mod presence.  Of course if you get ahold of Draylar he would know better than I.  :)

**Comment by NoComment1105** [comment]
Okay thanks

**Comment by NoComment1105** [comment]
I am having difficulty implementing maybe-data. Unless im reading it wrong, items should work but it wont work for me, after many atempts. The mod author is near unreachable so unfortunately this  issue may stay around for a while

**Comment by gniftygnome** [comment]
Created PR #3 for this.  Hope it helps.  Sorry it took a while; I've been busy lately.

**Comment by NoComment1105** [comment]
All fixed in versions 1.2.2
